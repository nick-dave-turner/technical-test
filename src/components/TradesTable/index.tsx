import React from "react";

import Table from "../Table";
import Widget from "../Widget";

type TradesTableProps = {
  rowData: Array<ITrade>;
};

const TradesTable: React.FC<TradesTableProps> = ({ rowData }) => {
  const columns = React.useMemo(
    () => [
      {
        Header: "Stock",
        accessor: "stockSymbol"
      },
      {
        Header: "Price",
        accessor: "price"
      },
      {
        Header: "No. Shares Purchased",
        accessor: "quantity"
      },

      {
        Header: "Transaction Date",
        accessor: "timestamp"
      }
    ],
    []
  );

  const data = React.useMemo(() => rowData, [rowData]);

  return (
    <Widget title="Recent Trades">
      <Table columns={columns} data={data} />
    </Widget>
  );
};

export default TradesTable;
