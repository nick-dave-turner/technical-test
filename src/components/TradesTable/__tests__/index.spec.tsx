import React from "react";
import { shallow } from "enzyme";

import TradesTable from "../";

const colDefs = [
  {
    Header: "Stock",
    accessor: "stockSymbol"
  },
  {
    Header: "Price",
    accessor: "price"
  },
  {
    Header: "No. Shares Purchased",
    accessor: "quantity"
  },

  {
    Header: "Transaction Date",
    accessor: "timestamp"
  }
];

const rowData = [
  {
    stockSymbol: "stockSymbol",
    price: 100,
    quantity: 2000,
    timestamp: "09/03/2020"
  }
];

describe("TradesTable component", () => {
  describe("When required props are empty", () => {
    it("should match snapshot", () => {
      const component = shallow(<TradesTable rowData={[]} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe("When required props are complete", () => {
    it("should match snapshot", () => {
      const component = shallow(<TradesTable rowData={rowData} />);
      expect(component).toMatchSnapshot();
    });
  });
});
