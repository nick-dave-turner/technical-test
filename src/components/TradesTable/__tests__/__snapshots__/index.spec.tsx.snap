// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradesTable component When required props are complete should match snapshot 1`] = `
<Widget
  title="Recent Trades"
>
  <Table
    columns={
      Array [
        Object {
          "Header": "Stock",
          "accessor": "stockSymbol",
        },
        Object {
          "Header": "Price",
          "accessor": "price",
        },
        Object {
          "Header": "No. Shares Purchased",
          "accessor": "quantity",
        },
        Object {
          "Header": "Transaction Date",
          "accessor": "timestamp",
        },
      ]
    }
    data={
      Array [
        Object {
          "price": 100,
          "quantity": 2000,
          "stockSymbol": "stockSymbol",
          "timestamp": "09/03/2020",
        },
      ]
    }
  />
</Widget>
`;

exports[`TradesTable component When required props are empty should match snapshot 1`] = `
<Widget
  title="Recent Trades"
>
  <Table
    columns={
      Array [
        Object {
          "Header": "Stock",
          "accessor": "stockSymbol",
        },
        Object {
          "Header": "Price",
          "accessor": "price",
        },
        Object {
          "Header": "No. Shares Purchased",
          "accessor": "quantity",
        },
        Object {
          "Header": "Transaction Date",
          "accessor": "timestamp",
        },
      ]
    }
    data={Array []}
  />
</Widget>
`;
